components/reddit/reddit.app.js

```diff
@@ -0,0 +1,189 @@
+const axios = require("axios");
+const get = require("lodash.get");
+module.exports = {
+  type: "app",
+  app: "reddit",
+  propDefinitions: {
+    subreddit: {
+      type: "string",
+      label: "Subreddit",
+      description: "The subreddit you'd like to watch.",
+      optional: false,
+      /* #region  subreddit prop's async options method     */
+      useQuery: true,
+      async options(context) {
+        const q = context.query;
+        const options = [];
+        const results = await this.getAllSearchSubredditsResults(q);
+        for (const subreddit of results) {
+          options.push({
+            label: subreddit.title,
+            value: subreddit.name,
+          });
+        }
+        return options;
+      },
+      /* #endregion */
+    },
+    username: {
+      type: "string",
+      label: "Username",
+      description: "The username you'd like to watch.",
+    },
+  },
+  methods: {
+    _accessToken() {
+      return this.$auth.oauth_access_token;
+    },
+    _apiUrl() {
+      return `https://oauth.reddit.com`;
+    },
+    async _makeRequest(opts) {
+      if (!opts.headers) opts.headers = {};
+      opts.headers.authorization = `Bearer ${this._accessToken()}`;
+      opts.headers["user-agent"] = "@PipedreamHQ/pipedream v0.1";
+      const { path } = opts;
+      delete opts.path;
+      opts.url = `${this._apiUrl()}${path[0] === "/" ? "" : "/"}${path}`;
+      return (await axios(opts)).data;
+    },
+    wereThingsPulled(reddit_things) {
+      const things = get(reddit_things, "data.children");
+      return things && things.length;
+    },
+    did4xxErrorOccurred(err) {
+      return (
+        get(err, "response.status") !== undefined &&
+        get(err, "response.status") !== null &&
+        err.response.status >= 400
+      );
+    },
+    async getNewHotSubredditPosts(subreddit, g, show, sr_detail, limit = 100) {
+      const params = {};
+      if (show) {
+        params["show"] = "all";
+      }
+      params["g"] = g;
+      params["sr_detail"] = sr_detail;
+      params["limit"] = limit;
+      return await this._makeRequest({
+        path: `/r/${subreddit}/hot`,
+        params,
+      });
+    },
+    async getNewSubredditLinks(before_link, subreddit, limit = 100) {
+      return await this._makeRequest({
+        path: `/r/${subreddit}/new`,
+        params: {
+          before: before_link,
+          limit,
+        },
+      });
+    },
+    async getNewSubredditComments(
+      subreddit,
+      subreddit_id36,
+      context,
+      depth,
+      sr_detail,
+      limit = 100
+    ) {
+      const resp = await this._makeRequest({
+        path: `/r/${subreddit}/comments/article`,
+        params: {
+          article: subreddit_id36,
+          context,
+          depth,
+          limit,
+          sort: "new",
+          sr_detail,
+          theme: "default",
+          threaded: true,
+          trucate: 0,
+        },
+      });
+      return resp[1];
```

Comment:
What does `resp[0]` contain? It might be nice to add a comment here indicating the structure of the `resp` array, and why you're returning the second element.

Reply:
So resp[0] contains the post (or article, as it is named in the corresponding Reddit API endpoint parameter) details where the comment was posted on.

resp[1] contains the new comments.

in this scnreeshot taken from a pd workflow, you will see the kind "t3" with only a child (the article parent to the comments) t3 is for link or post.
you will also see "t1" for comments, with 4 children (4 new comments found)/
https://sergio_eliot-hotmail.tinytake.com/tt/NTIwODUwOV8xNjI5NDk0OA

Reply:
Can you do this then?
```js
const [
  article,
  comments,
] = await this._makeRequest({
  // ...
});
return comments;
```

Also, seems like the `sort: "new"` parameter will not filter old comments, it will just sort the comments from newest to oldest. Shouldn't we have a `before`/`after` parameter here?

Reply:
Yes, totally, with much descriptive variable naming.

Reply:
That particular Reddit API call doesn't expect the usual `before` nor `after` parameters for pagination. Rather, it returns the whole "comment tree" whose dimensions are set by `depth` and `context` parameters.